Automattic/wp-calypso
# Import flow: Update UpgradePlan step, remove trial acknowledge step from import flows

Closes https://github.com/Automattic/wp-calypso/issues/82920

## Proposed Changes

* Removed obsolete (non-used) hooks: `use-checkout-url.ts`, `use-signup-step-navigator.ts`
* Simplified the way of building checkout URL based on siteSlug and planSlug;
* Redesigned the Upgrade plan step
* Introduced monthly/annual plan pricing switcher
* Remove the Trial Acknowledge step from the import flows

## Testing Instructions

@Automattic/caribou, please test as many scenarios as you can since there were a lot of changes.

* Try any of those three flows: "import-focused", "site-setup", "import-hosted-site"
* Entry URL: `/setup/{flow-name}/import?siteSlug={TARGET_SITE_SLUG}`
* Try different scenarios:
  * Target-> simple site; Source -> JN jetpack non-connected
    * Check if there is a "Upgrade plan" screen
    * After the "plan upgrade" or "7 days trial"
    * Check if there is a "Install Jetpack" screen
    * Pass through the whole flow
    * 
  * Target-> simple site; Source -> JN jetpack connected
    * Check if there is "Upgarde plan" screen
    * Check if there is "...7 days trial" option
    * Try the Trial flow
    * Check the transition between steps
    * 
   * Try one of those scenarios with a non-verified user
     * Pass through the whole flow with the email verification step

**NOTE:** e2e and unit tests have been updated to match the design changes and remove the Trial Acknowledge step from the flows.

## Screenshots
<img width="658" alt="Screenshot 2023-10-30 at 13 13 52" src="https://github.com/Automattic/wp-calypso/assets/1241413/efce98bb-db3f-494e-848b-e2e827da5d6d">

<img width="523" alt="Screenshot 2023-10-30 at 13 14 24" src="https://github.com/Automattic/wp-calypso/assets/1241413/b0ce8985-72bc-4252-b4e3-dc2d686c0d7c">

<img width="623" alt="Screenshot 2023-11-07 at 22 35 20" src="https://github.com/Automattic/wp-calypso/assets/1241413/7e55e0cb-d771-434f-895e-243548ae5adc">


## Pre-merge Checklist
